### ISSUE TITLE
When equal_occupants is set to true and everyone is affiliated as member, deleting a room via REST admin APIs always returns error

### ISSUE THREAD

**Issue by apontini** [esl/MongooseIM#3472]
MongooseIM version: 5.0.0
Installed from: source
Erlang/OTP version: Erlang/OTP 23 [erts-11.2.2.8]

If I set every member as an equal occupant using the `equal_occupant` flag in the muc lights configurations, no one is able to delete rooms. This is due to the fact that everyone is affiliated as a member and the `delete_room` function in the `mod_muc_light_commands ` module requires an owner.

I think that the error is the in the following lines (240-242 mod_muc_light_commands.erl): 

```erlang
    case muc_light_room_name_to_jid_and_aff(OwnerJID, RoomName, DomainName) of
        {ok, RoomJID, owner} -> mod_muc_light:delete_room(jid:to_lus(RoomJID));
        {ok, _, _} -> {error, denied, "you can not delete this room"};
        ...
```

**Comment by apontini** [comment]
I just opened a PR to try and fix this. I'm by no means an Erlang developer, please perdon any heresy :)

**Comment by apontini** [comment]
Anyone looking at this?

**Comment by chrzaszcz** [comment]
We are developing new [GrapQL API](https://github.com/esl/MongooseIM/tree/feature/graphql) now. This issue will not be present in the new API anymore. Do you still need this feature in the REST API?

**Comment by apontini** [comment]
As I said in my other issue, yes, it would be useful. I'll check the PR I've opened ASAP.

**Comment by apontini** [comment]
@chrzaszcz I acceppted your suggestions on the opened PR

**Comment by chrzaszcz** [comment]
@apontini how about the solution in #3576? I think it is cleaner, but it is merged to a feature branch, which will be merged to master quite soon.

**Comment by apontini** [comment]
@chrzaszcz sure! Looks good to me

**Comment by gustawlippa** [comment]
PR with the feature branch with the fix: #3515 (work in progress for now).
